### PR TITLE
Added apr_interest function to financial

### PR DESCRIPTION
**Pull request by smturro2**
### Describe your change:

In almost every scenario financial institutions will give the APR for a loan or credit card. This function allows you to calculate the interest without having to calculate the effective interest rate from the APR.

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by smturro2** [comment]
Hi, is there additional changes that need to be made to proceed with the pull request? Thanks!

**Review by tianyizheng02 — APPROVED** [review]
@cclauss I think this is good to merge as long as it passes the current workflows

**Comment by tianyizheng02** [comment]
```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

financial/interest.py:113: error: Argument 3 to "compound_interest" has incompatible type "float"; expected "int"  [arg-type]
Found 1 error in 1 file (checked 1192 source files)
```
pre-commit is failing because of a mypy error in a different function in the same file

File: financial/interest.py [hunk 1]
```diff
@@ -77,6 +77,43 @@ def compound_interest(
...
+    """
+    if number_of_years <= 0:
+        raise ValueError("number_of_years must be > 0")
+    if nominal_annual_percentage_rate < 0:
+        raise ValueError("nominal_annual_percentage_rate must be >= 0")
+    if principal <= 0:
+        raise ValueError("principal must be > 0")
+
+    return compound_interest(
+        principal, nominal_annual_percentage_rate / 365, number_of_years * 365
```
Comment (cclauss) [inline financial/interest.py:113]:
```suggestion
        principal, nominal_annual_percentage_rate / 365, number_of_years * 365.0
```

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
The `compound_interest` function call in `apr_interest` is causing `mypy` to fail because `compound_interest` takes an `int` for the number of compounding periods:
```py
def compound_interest(
    principal: float,
    nominal_annual_interest_rate_percentage: float,
    number_of_compounding_periods: int,
) -> float:
```
We should change the type hint for `number_of_compounding_periods` from `int` to `float`.

**Review by cclauss — APPROVED** [review]

**Pull request merged**